flocx_market/resources/offer.py

```diff
@@ -0,0 +1,88 @@
+from flask_restful import Resource, reqparse
+from models.offer import OfferModel
+
+
+class Offer(Resource):
+    parser = reqparse.RequestParser()
+    parser.add_argument('marketplace_offer_id',
+                        required=True,
+                        help="marketplace_offer_id needed"
+                       )
+    parser.add_argument('provider_id',
+                        required=True,
+                        help="provider_id needed"
+                        )
+    parser.add_argument('creator_id',
+                        required=True,
+                        help="creator_id needed"
+                        )
+    parser.add_argument('marketplace_date_created',
+                        required=True,
+                        help="marketplace_date_created needed"
+                        )
+    parser.add_argument('status',
+                        required=True,
+                        help="status needed"
+                        )
+    parser.add_argument('server_id',
+                        required=True,
+                        help="server_id needed"
+                        )
+    parser.add_argument('start_time',
+                        required=True,
+                        help="start_time needed"
+                        )
+    parser.add_argument('end_time',
+                        required=True,
+                        help="end_time needed"
+                        )
+    parser.add_argument('server_config',
+                        required=True,
+                        help="server_config needed"
+                        )
+    parser.add_argument('cost',
+                        required=True,
+                        help="cost needed"
+                        )
+
+    @classmethod
+    def get(cls, marketplace_offer_id):
+        offer = OfferModel.find_by_id(marketplace_offer_id)
+        if offer:
+            return offer.json()
+        return {'message': 'Offer not found'}, 404
+
+    @classmethod
+    def post(cls):
```

Comment:
Maybe this should be called `create`?

Reply:
this is the api endpoint which calls the create/save_to_db()
I think since it extends resource, it has to follow the REST convention: get, post, put, etc.

Reply:
If that's the case, you can ignore this (and the following) comment.